tests/run-pass/adjacent-allocs.rs

```diff
@@ -0,0 +1,17 @@
+/// Calling this function triggers a Miri allocation that gets deallocated.
```

Comment:
Please reference the relevant comment in #1866 to give some more context.

Reply:
Sure.  This seems pretty hacky to me anyway, is there a better way to trigger a Miri allocation that will get deallocated, and has a reasonable chance of being allocated next to `n`?

I was hoping that something like `let m = 0u64; drop(m)` would work but it doesn't.

Reply:
Oh I can just use a nested scope.

Reply:
You could create a heap allocation using `Box` and then `drop` it.

Reply:
I tried that before but it's impossible for two consecutive `Box::new()`s to get consecutive addresses, too many other things are allocated in between.  An interesting side effect of the address space being totally un-segmented between stack, heap, and global allocations :)

Reply:
Ah, yes that makes sense.
Please just leave a note explaining this in the code. :)

Reply:
Hopefully this is a good explanation?  https://github.com/rust-lang/miri/pull/1930/files#diff-92fc500453de24b690ac9cd35b3de8ca9c0d6e30d8bb86519114b889c31e1e49R8-R11